e2e/src/edit.spec.ts

```diff
@@ -0,0 +1,97 @@
+}
+
+test('Drag-and-drop pages via mouse interaction', async ({ context, page }) => {
+  const key = '62ba7264-e869-40fc-ac68-5892f1228a9b';
+  const obj = {
+    "summary": {
+      "title": "My form - 2024-07-10T20:20:08.037Z",
+      "description": ""
+    },
+    "root": "root",
+    "patterns": {
+      "root": {
+        "type": "page-set",
+        "id": "root",
+        "data": {
+          "pages": [
+            "cbd810ef-fd29-48f6-901b-fa88ab4f4100",
+            "5e4381b5-c05e-4471-ba1c-cd4ff045ab74",
+            "aca1a6a4-e7ba-4f37-bdf1-2726aa40831a"
+          ]
+        }
+      },
+      "cbd810ef-fd29-48f6-901b-fa88ab4f4100": {
+        "type": "page",
+        "id": "cbd810ef-fd29-48f6-901b-fa88ab4f4100",
+        "data": {
+          "title": "Page 1",
+          "patterns": []
+        }
+      },
+      "5e4381b5-c05e-4471-ba1c-cd4ff045ab74": {
+        "id": "5e4381b5-c05e-4471-ba1c-cd4ff045ab74",
+        "type": "page",
+        "data": {
+          "title": "A new page",
+          "patterns": []
+        }
+      },
+      "aca1a6a4-e7ba-4f37-bdf1-2726aa40831a": {
+        "id": "aca1a6a4-e7ba-4f37-bdf1-2726aa40831a",
+        "type": "page",
+        "data": {
+          "title": "Different Page",
+          "patterns": []
+        }
+      }
+    },
+    "outputs": []
+  };
+  const value = JSON.stringify(obj);
+
+  await context.addInitScript(([key, value]) => {
+    localStorage.setItem(key, value);
+  }, [key, value])
+
+  const pageTitles = Object.values(obj.patterns).filter(item => {
+    return item.type === 'page';
+  }).map(item => {
+    return (item.data as PageDataTest).title;
+  });
+
+  await page.goto(`${BASE_URL}`);
+  await page.getByRole('link', { name: 'Edit' }).click();
+  const buttonText = 'Move this item';
+
+  const firstUrl = new URL(page.url());
+  expect(firstUrl.hash.indexOf('?page=')).toEqual(-1);
+
+  const handle = page.locator('li').filter({ hasText: `${buttonText}${pageTitles[0]}` }).getByRole('button');
+  await handle.hover();
+  await page.mouse.down();
+  const nextElement = page.locator('li').filter({ hasText: `${buttonText}${pageTitles[1]}` }).getByRole('button');
+  await nextElement.hover();
+  await page.mouse.up();
+
+  await page.waitForFunction(([pageTitles, buttonText]) => {
+    const items = document.querySelectorAll('.usa-sidenav .draggable-list-item-wrapper');
+    return (items[0] as HTMLElement).innerText === buttonText + '\n' + pageTitles[1] && items.length === 3;
+  }, [pageTitles, buttonText]);
+  const pageTitlesCopy = [...pageTitles];
+  const newFirstItem = pageTitlesCopy.shift();
+
+  const newPageTitles = pageTitlesCopy.toSpliced(1, 0, newFirstItem || '');
+  const reorderedFirst = page.locator('ul').filter({ hasText: buttonText + newPageTitles.join(buttonText) }).getByRole('button').first();
+
+  await expect(reorderedFirst).toBeVisible();
+
+  const nextUrl = new URL(page.url());
+  expect(nextUrl.hash.indexOf('?page=1')).not.toEqual(-1);
+
+});
```

Comment:
I don't have much experience with this, but I wonder if something like a Page Object Model (POM) would help make this code easier to understand. https://playwright.dev/docs/pom